### PR TITLE
chore: use pipeable operators

### PR DESCRIPTION
**Pull request by jnizet**
It was easier than I thought to migrate: I didn't have inference problems as I did in my early experiments. 

I find the code less readable than before: `of(...)` is leass readable than `Observable.of(...)`.

I chose to use `ErrorObservable.create(...)` rather than `_throw(...)`, which really doesn't look nice, IMO.

The bundle size doesn't seem to be any different. But there is an advantage: it's now impossible to have missing or redundant imports. Missing imports cause a compilation error, and redundant imports, thanks to the strict checks we added before on the cmpiler, are now signalled by an error, too.

### CODE REVIEW

File: frontend/src/app/max-validator.directive.spec.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* tslint:disable:deprecation (false positive) */
```
Comment (cexbrayat) [inline frontend/src/app/max-validator.directive.spec.ts]: Is this related?

File: frontend/src/app/password-change/password-change.component.spec.ts [hunk 2]
```diff
@@ -117,7 +116,7 @@ describe('PasswordChangeComponent', () => {
     fixture.detectChanges();
 
     const userService = TestBed.get(CurrentUserService);
-    spyOn(userService, 'checkPassword').and.returnValue(Observable.throw(null));
+    spyOn(userService, 'checkPassword').and.returnValue(ErrorObservable.create(null));
```
Comment (cexbrayat) [inline frontend/src/app/password-change/password-change.component.spec.ts]: I get your point, but in the upcoming RxJS 6 it will be renamed `throwError` which is nicer, and I think importing `ErrorObservable` might be impossible.
Quoting from the [changelog](https://github.com/ReactiveX/rxjs/blob/master/CHANGELOG.md#breaking-changes):
```
* Observable: You should no longer deep import custom Observable implementations such as ArrayObservable or ForkJoinObservable.
* _throw: _throw is now exported as throwError
```

File: frontend/src/app/max-validator.directive.spec.ts (hunk 1 above)
Comment (jnizet) [reply frontend/src/app/max-validator.directive.spec.ts]: No. I must have missed it the last time I made an upgrade.

File: frontend/src/app/password-change/password-change.component.spec.ts (hunk 2 above)
Comment (jnizet) [reply frontend/src/app/password-change/password-change.component.spec.ts]: I'll change it when RxJS6 comes out, then. We don't use it much, and it will be easy to change.

**Comment by jnizet** [comment]
@cexbrayat I added the tslint rule: https://github.com/Ninja-Squad/globe42/blob/25bb7dbc8ae9c01c73a649751c06737e83bd35dd/frontend/tslint.json#L120

The excludes are for the test.ts and polyfills.ts files, which need those imports.
